File: JavaScript/452-Minimum Number of Arrows to Burst Balloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    points.sort(function(a,b){
        return a[1]-b[1];
    })
    
    let res = 0;
    let lastEnd = Number.NEGATIVE_INFINITY;
    
    for (let [start, end] of points){
        if (start > lastEnd){
            res++;
            lastEnd = end;
        }
    }
    
    return res;
};

File: C++/452-Minimum Number of Arrows to Burst Balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end());
        int N = points.size();
        int res = 1;
        
        int firstEnd = points[0][1];
        for (int i=0; i<N; i++) {
            int s = points[i][0], e = points[i][1];
            if (s <= firstEnd) {
                firstEnd = min(firstEnd, points[i][1]);
            } else {
                res++;
                firstEnd = e;
            }
        }
        return res;
    }
};